### DISCUSSION TITLE
Start publishing source releases again

### DISCUSSION THREAD

**Discussion by CyberTailor** [RHVoice/RHVoice#839] [General]
It's really hard to package a project if it uses lots of submodules. That's why v1.8.0 is still the last version in the repos:

https://repology.org/project/rhvoice/versions

Please provide complete source tarballs so distro maintainers can pick them up.

**Comment by Olga-Yakovleva** [comment]
Hello! Sorry for the late reply. Do you mean a single tarball with all the submodules? If so, this doesn't make sense for RHVoice. It doesn't have a single version progression for everything. Languages and voices may be updated at any point, and the core engine will remain the same. Languages may be updated and their voices will keep working with newer language versions. And a voice may be updated, but everything else stays the same. New languages may be added, and they may be compatible with the older cores, or need the latest version with its new features added to support this language. New voices for existing languages may be added.

Ideally, languages and voices should be packaged by distros as separate packages: a core language package, language packages depending on the core, voice packages depending on their corresponding language packages. I imagine it would be much work for maintainers to set it all up initially, but this would fit the structure of RHVoice. Or, if somebody can help with the internal voice manager I created for the snap, see what if anything need to be done to make it possible to include it in distro packages of the RHVoice core, we could stop including voices and languages in packages for distributions, and ask users to manage voices using this new command.

**Comment by CyberTailor** [comment]
Thanks for your reply.

I'll do languages/voice packaging separately (it'd be easy to do in Gentoo). However, there are build system components (`cmake/thirdParty`) that are included as submodules.

Ideally, they should be included in one tarball or bundled.
